### PR TITLE
Revert 35 revert 33 build docker for e2e

### PR DESCRIPTION
**Pull request by harrryr**
*Issue #, if available:*
Had to revert this [PR](https://github.com/aws-observability/aws-application-signals-test-framework/pull/33) due to syntax error in the publish metric step. Reposting the change with error fixed

Test run: https://github.com/aws-observability/aws-application-signals-test-framework/actions/runs/8621464546
Also checked that the metric is present in the console
Test run after switching to ECR: https://github.com/aws-observability/aws-application-signals-test-framework/actions/runs/8644705184

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/appsignals-e2e-ec2-test.yml [hunk 1]
```diff
@@ -31,13 +31,17 @@ env:
   LOG_GROUP_NAME: /aws/appsignals/generic
   TEST: ${{ inputs.test }}
   GET_ADOT_JAR_COMMAND: "wget -O adot.jar https://github.com/aws-observability/aws-otel-java-instrumentation/releases/latest/download/aws-opentelemetry-agent.jar"
+  TEST_RESOURCES_FOLDER: /__w/aws-application-signals-test-framework/aws-application-signals-test-framework
```
Comment (zzhlogin) [inline .github/workflows/appsignals-e2e-ec2-test.yml:34]: Why are we changing `/home/runner/work` to be `/__w/`? what does `/__w/` mean?

**Review by zzhlogin — COMMENTED** [review]
I have a general question regarding using container for Github workflow:
The goal of E2E tests is making sure the ADOT Java is working as expected in both EC2 and EKS, which mimic customer behaviours while using ADOT Java. In this PR, we are creating container for workflow runs, does this mean any steps defined within that job will be executed within the context of this container? In this case, we are not running everything from scratch in EC2/EKS, where customer might not do the same. Will this conflict the original goal?

**Comment by majanjua-amzn** [comment]
> I have a general question regarding using container for Github workflow: The goal of E2E tests is making sure the ADOT Java is working as expected in both EC2 and EKS, which mimic customer behaviours while using ADOT Java. In this PR, we are creating container for workflow runs, does this mean any steps defined within that job will be executed within the context of this container? In this case, we are not running everything from scratch in EC2/EKS, where customer might not do the same. Will this conflict the original goal?

@zzhlogin the container Harry is adding here only installs things that we can assume customers will have installed already and without error, for example, git, eksctl, kubectl, awscli, as they would not be able to interact with their EKS cluster or EC2 instance effectively without them. These packages are also not core to what we are trying to test, which is the enablement of Application Signals on these platforms. Checking whether kubectl can be installed is not part of the scope of the test and has only been a blocker/transient issue that stops the actual test from running.

File: .github/workflows/appsignals-e2e-ec2-test.yml (hunk 1 above)
Comment (harrryr) [reply .github/workflows/appsignals-e2e-ec2-test.yml:34]: That seems to be the default directory of the docker image we are using as the environment

**Review by zzhlogin — APPROVED** [review]

**Review by majanjua-amzn — APPROVED** [review]
Confirmed offline with Harry that a test run with a forced failure still behaves appropriately and sends failure metrics for our alarms.

**Pull request merged**